### DISCUSSION TITLE
[Nginx API Gateway] Change miauserproperties header name

### DISCUSSION THREAD

**Discussion by malta895** [mia-platform/community#253] [Q&A] [answered]
### Which section?

Console

### Which is the version?

9.5.3

### Description

Hello,
I need to change the name of the header `miauserproperties`, sent by the Api Gateway to the services. 
I found its definition in the Configuration repo of the project, in the generated file `auth-set-userid.conf`:

![image](https://github.com/mia-platform/community/assets/27235626/892b4413-a23f-45f7-8a59-c8c9709dbb8c)

I think that to customize the name I have to rewrite it in one of  the files in the *Advanced* section of the project, with a line like the following: 
```nginx
proxy_set_header 'custom-miauserproperties-header-name' $mia_userproperties;
```
Am I wrong? Which is the right file where to put this line?

Thanks!

**Comment by davidebianchi** [comment, marked as answer]
Hi @malta895! You can overwrite the headers set in the location of the api-gateway using the `root-location-extension.conf` file.
Here you can add the proxy of the header, as written in your description:

```nginx
proxy_set_header 'custom-miauserproperties-header-name' $mia_userproperties;
```

Also, it should be possible to hide the header with

```nginx
proxy_hide_header 'miauserproperties';
```
